WooCommerce/Classes/ViewRelated/Orders/Order Filters/FilterOrderListViewModel.swift

```diff
@@ -0,0 +1,120 @@
+    }
+
+    var criteria: Filters {
+        let orderStatus = orderStatusFilterViewModel.selectedValue as? OrderStatusEnum ?? nil
+        let dateRange = dateRangeFilterViewModel.selectedValue as? OrderDateRangeFilterEnum ?? nil
+        let numberOfActiveFilters = filterTypeViewModels.numberOfActiveFilters
+        return Filters(orderStatus: orderStatus,
+                       dateRange: dateRange,
+                       numberOfActiveFilters: numberOfActiveFilters)
+    }
+
+    func clearAll() {
+        let clearedOrderStatus: OrderStatusEnum? = nil
+        orderStatusFilterViewModel.selectedValue = clearedOrderStatus
+
+        let clearedDateRange: OrderDateRangeFilterEnum? = nil
+        dateRangeFilterViewModel.selectedValue = clearedDateRange
+    }
+}
+
+extension FilterOrderListViewModel {
+    /// Rows listed in the order they appear on screen
+    ///
+    enum OrderListFilter {
+        case orderStatus
+        case dateRange
+    }
+}
+
+private extension FilterOrderListViewModel.OrderListFilter {
+    var title: String {
+        switch self {
+        case .orderStatus:
+            return Localization.rowTitleOrderStatus
+        case .dateRange:
+            return Localization.rowTitleDateRange
+        }
+    }
+}
+
+extension FilterOrderListViewModel.OrderListFilter {
+    func createViewModel(filters: FilterOrderListViewModel.Filters) -> FilterTypeViewModel {
+        switch self {
+        case .orderStatus:
+            let options: [OrderStatusEnum?] = [nil, .pending, .processing, .onHold, .failed, .cancelled, .completed, .refunded]
+            return FilterTypeViewModel(title: title,
+                                       listSelectorConfig: .staticOptions(options: options),
+                                       selectedValue: filters.orderStatus)
+        case .dateRange:
+            let options: [OrderDateRangeFilterEnum?] = [nil, .today, .last2Days, .thisWeek, .thisMonth, .custom]
+            return FilterTypeViewModel(title: title,
+                                       listSelectorConfig: .staticOptions(options: options),
+                                       selectedValue: filters.dateRange)
+        }
+    }
+}
+
+// MARK: - FilterType conformance
+extension OrderStatusEnum: FilterType {
+    var isActive: Bool {
+        return true
+    }
+}
+
+// MARK: - Constants
+private extension FilterOrderListViewModel {
+    enum Localization {
+        static let filterActionTitle = NSLocalizedString("Show Orders", comment: "Button title for applying filters to a list of orders.")
+    }
+}
+
+private extension FilterOrderListViewModel.OrderListFilter {
+    enum Localization {
+        static let rowTitleOrderStatus = NSLocalizedString("Order Status", comment: "Row title for filtering orders by order status.")
+        static let rowTitleDateRange = NSLocalizedString("Date range", comment: "Row title for filtering orders by date range.")
```

Comment:
nit: in the [design](https://user-images.githubusercontent.com/495617/137908436-c2b83fb9-aabc-461b-89da-dc59c1296dab.jpg), "range" is capitalized 😄 
```suggestion
        static let rowTitleDateRange = NSLocalizedString("Date Range", comment: "Row title for filtering orders by date range.")
```

Reply:
Nice catch, updated!